Scripts/build-webdriveragent.js

```diff
@@ -1,88 +1,55 @@
 
   // Clean and build
-  log.info('Running ./Scripts/build.sh');
-  let env = {TARGET: 'runner', SDK: 'sim'};
   try {
-    await exec('/bin/bash', ['./Scripts/build.sh'], {env, cwd: rootDir});
+    await exec('/bin/bash', ['./Scripts/build.sh'], {
+      env: {TARGET: 'runner', SDK: 'sim', DERIVED_DATA_PATH},
+      cwd: ROOT_DIR
+    });
   } catch (e) {
-    log.error(`===FAILED TO BUILD FOR ${xcodeVersion}`);
-    log.error(e.stdout);
-    log.error(e.stderr);
-    log.error(e.message);
+    LOG.error(`===FAILED TO BUILD FOR ${xcodeVersion}`);
+    LOG.error(e.stdout);
+    LOG.error(e.stderr);
+    LOG.error(e.message);
     throw e;
   }
 
-  // Create bundles folder
-  const pathToBundles = path.resolve(rootDir, 'bundles');
-  await mkdirp(pathToBundles);
-
-  // Start creating zip
-  const uncompressedDir = path.resolve(rootDir, 'uncompressed');
-  await fs.rimraf(uncompressedDir);
-  await mkdirp(uncompressedDir);
-  log.info('Creating zip');
-
-  // Move contents of the root to folder called "uncompressed"
-  await exec('rsync', [
-    '-av', '.', uncompressedDir,
-    '--exclude', 'node_modules',
-    '--exclude', 'build',
-    '--exclude', 'ci-jobs',
-    '--exclude', 'lib',
-    '--exclude', 'test',
-    '--exclude', 'bundles',
-    '--exclude', 'azure-templates',
-  ], {cwd: rootDir});
-
-  // Move DerivedData/WebDriverAgent-* from Library to "uncompressed" folder
-  const wdaPath = (await fs.glob(`${derivedDataPath}/WebDriverAgent-*`))[0];
-  await mkdirp(path.resolve(uncompressedDir, 'DerivedData'));
-  await fs.rename(wdaPath, path.resolve(uncompressedDir, 'DerivedData', 'WebDriverAgent'));
-
-  // Compress the "uncompressed" bundle as a Zip
-  const pathToZip = path.resolve(pathToBundles, `webdriveragent-xcode_${xcodeVersion}.zip`);
-  await zip.toArchive(
-    pathToZip, {cwd: path.join(rootDir, 'uncompressed')}
-  );
-  log.info(`Zip bundled at "${pathToZip}"`);
-
-  // Now just zip the .app and place it in the root directory
-  // This zip file will be published to NPM
-  const wdaAppBundle = 'WebDriverAgentRunner-Runner.app';
-  const appBundlePath = path.join(uncompressedDir, 'DerivedData', 'WebDriverAgent',
-    'Build', 'Products', 'Debug-iphonesimulator', wdaAppBundle);
-  const appBundleZipPath = path.join(rootDir, `${wdaAppBundle}.zip`);
+  const zipName = `WebDriverAgentRunner-Runner-Sim-${xcodeVersion}.zip`;
+  LOG.info(`Creating ${zipName} which includes ${WDA_BUNDLE}`);
+  const appBundleZipPath = path.join(ROOT_DIR, zipName);
   await fs.rimraf(appBundleZipPath);
-  log.info(`Created './${wdaAppBundle}.zip'`);
-  await zip.toArchive(appBundleZipPath, {cwd: appBundlePath});
-  log.info(`Zip bundled at "${appBundleZipPath}"`);
-  // Clean up the uncompressed directory
-  await fs.rimraf(uncompressedDir);
+  LOG.info(`Created './${zipName}'`);
+  try {
+    await exec('xattr', ['-cr', WDA_BUNDLE], {cwd: WDA_BUNDLE_PATH});
+    await exec('zip', ['-r', appBundleZipPath, WDA_BUNDLE], {cwd: WDA_BUNDLE_PATH});
```

Comment:
i would also run zip with `-q` as it is quite talkative by default and uses extra memory to keep all these logs